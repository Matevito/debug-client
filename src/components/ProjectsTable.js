import React from 'react'

import { Link } from "react-router-dom";
import {

} from "@mui/material";

const Row = ({project}) => {

}
export const ProjectsTable = ({ projects }) => {
    return (
        <>todo...</>
    )
}
