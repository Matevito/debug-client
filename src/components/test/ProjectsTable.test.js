import { MemoryRouter } from "react-router-dom";
// testing comps
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'

//tested component;
import { ProjectsTable } from "../ProjectsTable";

describe("ProjectsTable component", () => {
    let projects;
    beforeEach(() => {
        projects = [
            {
                title: "project1",
                teamLeader: "null",
                issues: 15,
                solvedIssues: 12,
                _id :"project1_id"
            },
            {
                title: "project2",
                teamLeader: "null",
                issues: 14,
                solvedIssues: 8,
                _id :"project2_id"
            },
            {
                title: "project3",
                teamLeader: "null",
                issues: 2,
                solvedIssues: 0,
                _id :"project3_id"
            },
        ];
    })
    test("renders a correct table", () => {
        
        const testTitle = "testing title";

        render(<ProjectsTable projects={projects} title={testTitle}/>,  {wrapper: MemoryRouter})
        expect(screen.getByText("testing title")).toBeInTheDocument()
        expect(screen.getByText("project1")).toBeInTheDocument()
        expect(screen.getByText("project2")).toBeInTheDocument()
        expect(screen.getByText("project3")).toBeInTheDocument()

    });
    test("check snapshot", () => {
        const testTitle = "snapshot test!";
        const { container } = render(<ProjectsTable projects={projects} title={testTitle}/>,  {wrapper: MemoryRouter})
        expect(container).toMatchSnapshot();
    })
});