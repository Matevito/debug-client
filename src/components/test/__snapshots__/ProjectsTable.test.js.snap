// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsTable component check snapshot 1`] = `
<div>
  <div
    class="MuiTableContainer-root css-1ofbc68-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-1w3qztj-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-1lhsthh-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ndpvdd-MuiTableCell-root"
            colspan="6"
            scope="col"
          >
            snapshot test!
          </th>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            colspan="2"
            scope="col"
          >
            PROJECT
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            TEAM LEADER
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            ISSUES
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            ISSUES SOLVED
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            LINK
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-n1y5x-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colspan="2"
          >
            project1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            <a
              href="/user/null"
              style="text-decoration: none;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            15
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            12
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            <a
              href="/project/project1_id"
              underline="hover"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BuildIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </a>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colspan="2"
          >
            project2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            <a
              href="/user/null"
              style="text-decoration: none;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            14
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            8
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            <a
              href="/project/project2_id"
              underline="hover"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BuildIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </a>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colspan="2"
          >
            project3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            <a
              href="/user/null"
              style="text-decoration: none;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            0
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
          >
            <a
              href="/project/project3_id"
              underline="hover"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BuildIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
